src/services/walletconnect/utils.ts

```diff
@@ -28,3 +30,38 @@ export const getSupportedChainIds = (configs: Array<ChainInfo>, params: Proposal
     })
     .map((chain) => chain.chainId)
 }
+
+export const getNamespaces = (
+  proposal: Web3WalletTypes.SessionProposal,
+  currentChainId: string,
+  safeAddress: string,
+) => {
+  // Most dApps require mainnet, but we aren't always on mainnet
+  // As workaround, we pretend include all required and optional chains with the Safe chainId
+  const requiredChains = proposal.params.requiredNamespaces[EIP155]?.chains || []
+  const optionalChains = proposal.params.optionalNamespaces[EIP155]?.chains || []
+
+  const supportedChainIds = [currentChainId].concat(
+    requiredChains.map(stripEip155Prefix),
+    optionalChains.map(stripEip155Prefix),
+  )
+
+  const eip155ChainIds = supportedChainIds.map(getEip155ChainId)
+  const eip155Accounts = eip155ChainIds.map((eip155ChainId) => `${eip155ChainId}:${safeAddress}`)
+
+  // Don't include optionalNamespaces methods/events
+  const methods = proposal.params.requiredNamespaces[EIP155]?.methods ?? SAFE_COMPATIBLE_METHODS
+  const events = proposal.params.requiredNamespaces[EIP155]?.events || []
+
+  return buildApprovedNamespaces({
+    proposal: proposal.params,
+    supportedNamespaces: {
+      [EIP155]: {
+        chains: eip155ChainIds,
+        accounts: eip155Accounts,
+        methods,
+        events,
+      },
+    },
+  })
+}
```

Comment:
Stylistically, I don't see this function as a util. I would keep it as a private method in the WalletConnectWallet class.